assertj-core/src/main/java/org/assertj/core/api/AbstractClassAssert.java

```diff
@@ -211,6 +211,48 @@ public SELF isNotAnnotation() {
     return myself;
   }
 
+  /**
+   * Verifies that the actual {@code Class} is a record.
+   * <p>
+   * Example:
+   * <pre><code class='java'> public record Jedi(String name) {}
+   *
+   * // this assertion succeeds:
+   * assertThat(Jedi.class).isRecord();
+   *
+   * // this assertion fails:
+   * assertThat(String.class).isRecord();</code></pre>
+   *
+   * @return {@code this} assertions object
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} is not a record.
+   */
+  public SELF isRecord() {
+    classes.assertIsRecord(info, actual);
```

Comment:
No need to have the inner Classes layer as it's unlikely that this assertion will be shared somewhere else, please move the actual implementation here as a private method.